src/app/multi-editor/multi-editor.component.ts

```diff
@@ -61,23 +61,48 @@ export class MultiEditorComponent implements OnInit {
   }
 
   onSave() {
-    this.queryService.save(this.previewedActions, this.checkedRecords)
-    .then((res) => {
-      this.successMessage = res['message'];
-      this.totalRecords = -1;
-      this.changeDetectorRef.markForCheck();
-    })
+    let checkedUUIDs: string[];
+    this.processAll ? checkedUUIDs = Object.keys(this.recordSelectionStatus)
+      .filter(key => !this.recordSelectionStatus[key]) :
+      checkedUUIDs = Object.keys(this.recordSelectionStatus)
+        .filter(key => this.recordSelectionStatus[key]);
+    this.queryService.save(this.previewedActions, checkedUUIDs, this.processAll)
       .catch((error) => {
         if (error.json().message) {
           this.totalRecords = -1;
           this.errorMessage = error.json().message;
-        }else {
+        } else {
           this.errorMessage = error;
         }
         this.changeDetectorRef.markForCheck();
       });
   }
 
+  selectAll() {
+    this.processAll = true;
+    Object.keys(this.recordSelectionStatus).forEach((item) => { this.recordSelectionStatus[item] = true; });
+    this.changeDetectorRef.markForCheck();
+  }
+
+  deselectAll() {
+    this.processAll = false;
+    Object.keys(this.recordSelectionStatus).forEach((item) => { this.recordSelectionStatus[item] = false; });
+    this.changeDetectorRef.markForCheck();
+  }
+
+  private setNewCheckboxes() {
+    if (this.recordSelectionStatus.hasOwnProperty(this.uuids[0])) {
+      return;
```

Comment:
this is redundant,
```
if (!this.recordSelectionStatus.hasOwnProperty(this.uuids[0])) {
... do the thing ...
}
```

Reply:
It's not redundant. If I remove this the checkbox selection of a page that the user visit's again will be overwritten.

Reply:
redundant part is not the check but having `if` and `else` block where `if` part is empty and the logic is in `else`